### ISSUE TITLE
`kubectl describe svc` bad string formatting of TargetPort

### ISSUE THREAD

**Issue by victorhaggqvist** [kubernetes/kubernetes#60984]
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

Run

    kubectl describe svc jaeger-collector

Got
```
Name:              jaeger-collector
Namespace:         default
Labels:            app=jaeger
                   chart=jaeger-0.3.5
                   component=collector
                   heritage=Tiller
                   jaeger-infra=collector-service
                   release=jaeger
Annotations:       <none>
Selector:          app=jaeger,component=collector,jaeger-infra=collector-pod,release=jaeger
Type:              ClusterIP
IP:                10.55.255.22
Port:              jaeger-collector-tchannel  14267/TCP
TargetPort:        %!d(string=tchannel)/TCP
Endpoints:         10.52.1.6:14267
Port:              jaeger-collector-http  14268/TCP
TargetPort:        %!d(string=http)/TCP
Endpoints:         10.52.1.6:14268
Port:              jaeger-collector-zipkin  9411/TCP
TargetPort:        %!d(string=zipkin)/TCP
Endpoints:         10.52.1.6:9411
Session Affinity:  None
Events:            <none>
```

**What you expected to happen**:

TargetPort to be correctly formatted

    TargetPort:       tchannel/TCP

**How to reproduce it (as minimally and precisely as possible)**:

Run `kubectl describe svc jaeger-collector`, any service with named ports would do I guess, here https://github.com/kubernetes/charts/blob/13f4902f412d822bdeeae8923d7df2c9aea7acd8/incubator/jaeger/README.md was used.

Output of TargetPort to be is formatted like this

    TargetPort:        %!d(string=tchannel)/TCP

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.6", GitCommit:"6260bb08c46c31eea6cb538b34a9ceb3e406689c", GitTreeState:"clean", BuildDate:"2017-12-21T06:34:11Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.7-gke.0", GitCommit:"192ccad06d24af9828cbf42330e1d915cb586406", GitTreeState:"clean", BuildDate:"2018-01-25T21:58:21Z", GoVersion:"go1.8.3b4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: GKE
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by mlmhl** [comment]
It seems that this is already fixed in 1.9 version.

**Comment by victorhaggqvist** [comment]
In that case I'll close this.
